packages/web/contexts/GameContext.tsx

```diff
@@ -0,0 +1,328 @@
+   */
+  const handleChoice = useCallback(
+    async (target: string): Promise<string | undefined> => {
+      try {
+        if (target) gameState(target);
+        const success = gameState() === target;
+        if (success) {
+          return target;
+        }
+        throw new Error('Game progression failed');
+      } catch (error) {
+        // add toast here
+        console.log(error);
+        return undefined;
+      }
+    },
+    [],
+  );
+
+  /** Increment number of elements a user has visited & store in localStorage
+   * This is used to trigger an achievement after so many elements have been visited
+   */
+  const visitedElements = (increment?: boolean): string => {
+    const visited: string | null = get('OnboardingGameVisitedElements');
+
+    if (increment) {
+      const incrementVisits = parseInt(visited ?? '0', 10) + 1;
+      set(
+        'OnboardingGameVisitedElements',
+        visited ? incrementVisits.toString() : `1`,
+      );
+      return incrementVisits.toLocaleString();
+    }
+
+    return visited ?? '0';
+  };
+
+  const resetGame = useCallback((): boolean => {
+    gameState(undefined, true);
+    const success = gameState() === null;
+
+    if (success) {
+      return true;
+    }
+    return false;
+  }, []);
+
+  const mintChiev = useCallback(
+    async (tokenId: bigint): Promise<any> => {
+      try {
+        if (address === undefined) await connect();
+        if (provider == null) throw new Error('Provider not set.');
+
+        const contractAddress = CONFIG.chievContractAddress;
+        const token = new Contract(contractAddress, ABI, provider.getSigner());
+        const confirmationsWait = 2;
+        const metadataURI = await token.uri(tokenId);
+        if (!metadataURI || metadataURI === '') {
+          throw new Error(`No metadata for token ${tokenId}.`);
+        }
+        toast({
+          title: 'Claim in progress',
+          description: 'Please sign the transaction in your wallet.',
+          status: 'info',
+          isClosable: true,
+        });
+        const response = await fetch(httpLink(metadataURI)!);
+        const metadata = await response.json();
+        const tx = await token['mint(address[],uint256,bytes)'](
+          [address],
+          tokenId,
+          [],
+        );
+
+        toast({
+          title: 'Chiev claim',
+          description: (
+            <>
+              <Text>Transaction hash: {tx.hash}</Text>
+              <Text>Waiting for {confirmationsWait} confirmations&hellip;</Text>
+            </>
+          ),
+          status: 'info',
+          isClosable: true,
+          duration: 5000,
+        });
+
+        const receipt = await tx.wait(confirmationsWait);
+        const receiptUrl = `${CONFIG.polygonscanBaseURL}/tx/${tx.hash}`;
+        const nftURL = httpLink(metadata.external_url);
+
+        set('ChievClaimed', 'true');
+
+        toast({
+          title: 'Chiev claimed 🎉',
```

Comment:
I'd make the `title`s title case.

```suggestion
          title: '’Chiev Claimed 🎉',
```